source/common/http/conn_manager_impl.cc

```diff
@@ -1402,7 +1403,10 @@ void ConnectionManagerImpl::ActiveStream::onResetStream(StreamResetReason, absl:
   // DownstreamProtocolError and propagate the details upwards.
   const absl::string_view encoder_details = response_encoder_->getStream().responseDetails();
   if (!encoder_details.empty()) {
-    filter_manager_.streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DownstreamProtocolError);
+    if (reset_reason == StreamResetReason::LocalReset) {
```

Comment:
Maybe collapse this with the previous if block to reduce nesting? Just && the two conditions